specutils/io/default_loaders/twoslaq_lrg.py

```diff
@@ -0,0 +1,99 @@
+import astropy.io.fits as fits
+from astropy.units import Unit
+from astropy.wcs import WCS
+from specutils.io.registers import data_loader
+from specutils import Spectrum1D, SpectrumList
+
+
+def identify_2slaq_lrg(origin, *args, **kwargs):
+    """
+    Identify if the current file is a 2SLAQ-LRG file
+    """
+    with fits.open(args[0]) as hdulist:
+        if hdulist[0].header["MSTITLE"].startswith("2dF-SDSS LRG/QSO survey"):
+            # apparently the QSO part doesn't have MSTITLE, so we should be safe
+            # with just the above condition, but just in case, we know they have
+            # a different structure (LRG has one ext, QSO has more; LRG is 3d,
+            # QSO is 1d
+            if len(hdulist) == 1 and hdulist[0].data.ndim == 3:
+                return True
+        return False
+
+
+@data_loader("2SLAQ-LRG", identifier=identify_2slaq_lrg,dtype=SpectrumList,
+             extensions=["fit", "fits"])
+def twoslaq_lrg_fits_loader(filename, **kwargs):
+    """
+    Load a file from the LRG subset of the 2dF-SDSS LRG/QSO survey (2SLAQ-LRG)
+    file.
+
+    2SLAQ was one of a number of surveys that used the 2dF instrument on the
+    Anglo-Australian Telescope (AAT) at Siding Spring Observatory (SSO) near
+    Coonabarabran, Australia, and followed up the 2QZ survey. Further details
+    can be seen at http://www.physics.usyd.edu.au/2slaq/ or at
+    https://docs.datacentral.org.au/.
+
+    The LRG and QSO data appear to be in different formats, this only loads the
+    LRG version. As there is a science and sky spectrum, the `SpectrumList`
+    loader provides both, whereas the `Spectrum1D` loader only provides the
+    science.
+
+    Parameters
+    ----------
+    file_name: str
+        The path to the FITS file
+    Returns
+    -------
+    data: SpectrumList
+        The 2SLAQ-LRG spectrum that is represented by the data in this file.
+    """
+    with fits.open(filename) as hdulist:
```

Comment:
This should be updated to support the case (as in other loaders) of providing an already existing hdulist or file objects. See the [cos loader](https://github.com/astropy/specutils/blob/master/specutils/io/default_loaders/hst_cos.py#L38) as an example.